### ISSUE TITLE
Datetimepicker auto correcting year to 4 digit when i enter 2 digit

### ISSUE THREAD

**Issue by Liviano** [Eonasdan/bootstrap-datetimepicker#2162]
When we type a date value in the text field and if the year value is in 2 digit the plugin auto correct it to 4 digit...
when the value is greater that 68 there is no issues it changes it in to 1968 and so on, but when a value less than that is entered for example 56 it changes to 2056, but what i want to have is 1956...
is there any solution for this...
is there a way to stop the 4 digit formatting...

**Comment by William-H-M** [comment]
Hi can you show us a fiddle of what's going on?

**Comment by Liviano** [comment]
[http://eonasdan.github.io/bootstrap-datetimepicker/#minimum-setup](http://eonasdan.github.io/bootstrap-datetimepicker/#minimum-setup)

It happens in all the examples... type 121212.... it will auto format to 12/12/0012... i don't want it to auto format so that i can throw error and ask user to enter the year in 4 digit year format... if fiddle is mandate i can create one...

**Comment by William-H-M** [comment]
Ohhh I do understand now.
**tl;dr;**
You have two option not allow user to do whatever he wants or disable the option to write, [fiddle of both](https://jsfiddle.net/William_/k38sg9k2/)

Mmm will you mind or will damage the way you're using the picker currently if a tell you to use the picker of a different way?

If not, on my own experience things like dates is not a thing user should be allowed to type because of 2 simple things.
1) they don't exactly know the format with which your system works, even when it's a culture one!
2) as the don't know what exactly to type, this will take them more time, and can get or confusing or annoying

What I will recommend is a convination of 
```
  ignoreReadonly: true,
  allowInputToggle: true
```
thus the user will just click on the button of calendar, something that it's user friendly because most of them know that it'll appear a calendar popup, and they won't have to type anything, you can always set up viewMode to your prefference ofcourse.

But just in case you don't want to do what I said yo can alway do what was done on older issue
#1991, more exactly [this comment](https://github.com/Eonasdan/bootstrap-datetimepicker/issues/1991#issuecomment-274714378).

This I'll say it's the worst way because if a user has an space that seem you can write on it, and then seems like it's not appear nothing will thing it's broke an will leave, thing that we don't want rigth?!

**Comment by Liviano** [comment]
I agree with you @William-H-M 
I previously used read-only input... But as a part of requirement it is like either user can select date or they can enter a date in the input field... So after this only i started facing this problem... 
Please update if there is any other solution...
Thanks for your responds...

**Comment by William-H-M** [comment]
@Liviano Mmmm I've any idea on my mind but not sure if it'll work as I've never tried before, so you'll have to test it out, do you mind?

What I'd suggest is use one more plugin in combination with this one, use [RobinHerbots/Inputmask](https://github.com/RobinHerbots/Inputmask), I've used it before for some projects, but as I said before never in conjuction of this, just set the mask for a date that way they would be unable to write wrong date if they want to type.
And a good thing is that you can [validate ](https://github.com/RobinHerbots/Inputmask#isvalid) the masks.

What do you think about it?

**Comment by Liviano** [comment]
Thanks @William-H-M 
I have used another option.
useStrict: true,
keepInvalid: true
When i set this both values to true even when we enter a wrong value or format, the entered data remains in the input field... so that i can throw error and make the user to correct it...
Thanks for the help.

**Comment by William-H-M** [comment]
Great it was an even useful option I'll keep it on mind for later issues thax :D
